tooling/cli/src/bundle.rs

```diff
@@ -214,77 +229,72 @@ pub fn bundle<A: AppSettings>(
-        Ok(private_key) => {
-          // check if private_key points to a file...
-          let maybe_path = Path::new(&private_key);
-          let private_key = if maybe_path.exists() {
-            std::fs::read_to_string(maybe_path)?
-          } else {
-            private_key
-          };
-          updater_signature::secret_key(private_key, password)
-        }
-        _ => Err(anyhow::anyhow!("A public key has been found, but no private key. Make sure to set `TAURI_SIGNING_PRIVATE_KEY` environment variable.")),
-      }?;
-
-      let pubkey = base64::engine::general_purpose::STANDARD.decode(pubkey)?;
-      let pub_key_decoded = String::from_utf8_lossy(&pubkey);
-      let public_key = minisign::PublicKeyBox::from_string(&pub_key_decoded)?.into_public_key()?;
-
-      // make sure we have our package built
-      let mut signed_paths = Vec::new();
-      for bundle in update_enabled_bundles {
-        // we expect to have only one path in the vec but we iter if we add
-        // another type of updater package who require multiple file signature
-        for path in bundle.bundle_paths.iter() {
-          // sign our path from environment variables
-          let (signature_path, signature) = updater_signature::sign_file(&secret_key, path)?;
-          if signature.keynum() != public_key.keynum() {
-            log::warn!("The updater secret key from `TAURI_PRIVATE_KEY` does not match the public key from `plugins > updater > pubkey`. If you are not rotating keys, this means your configuration is wrong and won't be accepted at runtime when performing update.");
-          }
-          signed_paths.push(signature_path);
-        }
-      }
+  let Some(pubkey) = &update_settings.pubkey else {
+    // Maybe return error here?
+    return Ok(());
+  };
+  // get the public key
+  // check if pubkey points to a file...
+  let maybe_path = Path::new(&pubkey);
+  let pubkey = if maybe_path.exists() {
+    std::fs::read_to_string(maybe_path)?
+  } else {
+    pubkey.to_string()
+  };
 
-      print_signed_updater_archive(&signed_paths)?;
+  // if no password provided we use an empty string
+  let password = std::env::var("TAURI_SIGNING_PRIVATE_KEY_PASSWORD")
+    .ok()
+    .or_else(|| if ci { Some("".into()) } else { None });
+
+  // get the private key
+  let secret_key = match std::env::var("TAURI_SIGNING_PRIVATE_KEY") {
+            Ok(private_key) => {
+              // check if private_key points to a file...
+              let maybe_path = Path::new(&private_key);
+              let private_key = if maybe_path.exists() {
+                std::fs::read_to_string(maybe_path)?
+              } else {
+                private_key
+              };
+              updater_signature::secret_key(private_key, password)
+            }
+            _ => Err(anyhow::anyhow!("A public key has been found, but no private key. Make sure to set `TAURI_SIGNING_PRIVATE_KEY` environment variable.")),
+          }?;
```

Comment:
the formatting here is weird, let's format manually if `cargo fmt` fails to format it